modules/node_modules/@ncigdc/components/AddToCartButtonAll.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { css, style, hover } from 'glamor';
+import _ from 'lodash';
+
+import { parseFilterParam } from '@ncigdc/utils/uri';
+import LocationSubscriber from '@ncigdc/components/LocationSubscriber';
+import ShoppingCartIcon from 'react-icons/lib/fa/shopping-cart';
+import DownCaretIcon from 'react-icons/lib/fa/caret-down';
+import TrashIcon from 'react-icons/lib/fa/trash';
+
+import { addAllFilesInCart, fetchFiles, removeAllInCart, toggleFilesInCart } from '@ncigdc/dux/cart';
+
+import Row from '@ncigdc/uikit/Flex/Row';
+import Column from '@ncigdc/uikit/Flex/Column';
+import Dropdown from '@ncigdc/uikit/Dropdown';
+import Button from '@ncigdc/uikit/Button';
+
+import styled from '@ncigdc/theme/styled';
+
+const styles= {
+  cart: {
+    padding: '0 5px',
+    color: '#FFF',
+    borderRadius: '4px',
+    borderTopRightRadius: '0px',
+    borderBottomRightRadius: '0px',
+    cursor: 'pointer',
+  },
+  addToCart: {
+    backgroundColor: '#0d95a1',
+  },
+  removeFromCart: {
+    backgroundColor: '#255425',
+  }
+};
+
+const DropDownCaret = styled.span({
+  backgroundColor: '#FFF',
+  borderTopRightRadius: '4px',
+  borderBottomRightRadius: '4px',
+});
+
+const DropDownStyle = {
+  position: 'absolute',
+  right: 'initial',
+  left: '-25px',
+  marginTop: '5px',
+  borderRadius: '4px'
+};
+
+const DropDownItemStyle = {
+  padding: '3px 20px',
+  margin: '5px 0',
+  fontWeight: 'normal',
+}
+
+const filesInCart = (edges, files) => {
+  if(files.length < edges.length) return false;
+  else return _.every(edges, edge => _.some(files, {'file_id':edge.file_id}))
```

Comment:
every and some are both native to javascript. check the sidebar to see what you can use

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array